clang/test/Analysis/Checkers/WebKit/uncounted-local-vars.cpp

```diff
@@ -492,4 +492,42 @@ namespace virtual_function {
     auto* baz = &*obj;
     // expected-warning@-1{{Local variable 'baz' is uncounted and unsafe [alpha.webkit.UncountedLocalVarsChecker]}}
   }
-}
\ No newline at end of file
+}
+
+namespace vardecl_in_if_condition {
+  RefCountable* provide();
+
+  RefCountable* get() {
+    if (auto* obj = provide())
+      return obj; // no warning
+    return nullptr;
+  }
+
+  RefCountable* get_non_trivial_then() {
+    if (auto* obj = provide()) // expected-warning{{Local variable 'obj' is uncounted and unsafe [alpha.webkit.UncountedLocalVarsChecker]}}
+      return obj->next();
+    return nullptr;
+  }
+
+  RefCountable* get_non_trivial_else() {
+    if (auto* obj = provide())
+      return obj;
+    else
+      return provide()->next();
+    return nullptr;
+  }
```

Comment:
I was surprised to give special treatment to the "then" block. I wonder if the "else" block should be also considered.

```c++
    if (auto* obj = provide(); !obj)
      return nullptr;
    else
      return obj;
  }
```

In other words, and `if(condvar = init()) action()` can be represented by `if (condor = init(); !condvar) {/*empty*/} else action()`.

Reply:
But let me add that as a test case in the case someone changes this semantics in the future.

Reply:
But it's not really something to be fixed in that code since `getConditionVariable()` returns `nullptr` in this scenario, is it? Or maybe I'm misunderstanding what you're suggesting?